### DISCUSSION TITLE
[KMP] Undefined symbols Error with latest version

### DISCUSSION THREAD

**Discussion by BreakZero** [trustwallet/wallet-core#3424] [Q&A]
Hi, i am a new guy in iOS developer,  follow the sample, i setup my KMP Project, but i get an error. am i missing something in configuration? or do i need add a special linker in XCode? thanks. Btw, it work well in Android.

**Describe the bug**
error message:
```
ld: Undefined symbols:
  _TWCoinTypeDeriveAddressFromPublicKeyAndDerivation, referenced from:
      _com_trustwallet_core_TWCoinTypeDeriveAddressFromPublicKeyAndDerivation_wrapper39 in data[2](result.o)
clang: error: linker command failed with exit code 1 (use -v to see invocation)
```

**Additional context**
my gradle configuration file:
```
kotlin {
    cocoapods {
        dependencies {
            pod("TrustWalletCore", moduleName = "WalletCore")
        }
    }
    sourceSets {
        getByName("commonMain") {
            dependencies {
                api("com.trustwallet:wallet-core-kotlin:3.2.18")
            }
        }
     }
}
```

i also can see it in Pods via XCode Project
<img width="268" alt="image" src="https://github.com/trustwallet/wallet-core/assets/25579672/51a3b506-19ad-4e50-b726-80c45c273d7e">

**Comment by JaimeToca** [comment]
Hi @BreakZero, 
-> How does your iOS package looks like?
-> You can probably skip cocoapods dependency, and download specific one in the client

Ideally, as multiplatform project, you should create a new item as probably in the future you will need to adjust specific things to iOS platform,

```
val iosArm64Main by getting
val iosSimulatorArm64Main by getting
val iosX64Main by getting

create("iosMain") {
    dependsOn(<Dependencies>)
    iosArm64Main.dependsOn(this)
    iosSimulatorArm64Main.dependsOn(this)
    iosX64Main.dependsOn(this)
}
```

**Reply by BreakZero** [reply]
Hi @JaimeToca ,

My iOS platform gradle configuration like below, i use it as a plugin. When i comment the code where using `HDWallet`, then build and run success on iOS platform. So i think it is not an issue of gradle configuration.
<img width="1013" alt="image" src="https://github.com/trustwallet/wallet-core/assets/25579672/bbe0e76a-3702-4e36-81c2-fa2a6a1383f7">

BTW, just a doubt, i am on `XCode 15 Beta 5` and kotlin `1.9.10`, i known there are some changes in XCode 15 Beta that affect KMP. 
But from error log, it looks like a linker issue, not about XCode environment.

**Reply by BreakZero** [reply]
Hi @JaimeToca ,
I found the problem of this issue.
Here is a class 
```kotlin
class HDWalletInMemory : SynchronizedObject() {
    private var hdWallet: HDWallet? = null

    fun loadInMemory(
        mnemonic: String,
        passphrase: String = ""
    ) {
        synchronized(this) {
            hdWallet = HDWallet(mnemonic, passphrase)
        }
    }

    fun hdWallet(): HDWallet = synchronized(this) {
        checkNotNull(hdWallet)
    }

    fun release() {
        synchronized(this) {
            hdWallet = null
        }
    }
}
```
it will work well when i comment the function named `hdWallet()` which have a return type of `HDWallet`
So that means could not have any function return HDWallet. We need to create new one instance when we need it.

**Reply by JaimeToca** [reply]
Thanks @BreakZero we'll have a look at this
